### PR TITLE
Update build matrix variable interpolation

### PR DESCRIPTION
**Pull request by nsuma8989**

### CODE REVIEW

File: pages/pipelines/build_matrix.md.erb [hunk 1]
```diff
@@ -17,19 +17,24 @@ steps:
 ```
 {: codeblock-file="pipeline.yml"}
 
-You can use a Build Matrix to expand a single step into three steps by replacing the `{{matrix}}` variable in the label and command attributes of a [command step](/docs/pipelines/command-step) with each value provided in the `matrix` array. You can interpolate the matrix values into command steps (including step labels) and into [environment variables](/docs/pipelines/environment-variables).
+You can use a Build Matrix to expand a single step into three steps as you can interpolate the matrix values into [command step](/docs/pipelines/command-step) attributes such as [environment variables](/docs/pipelines/environment-variables), label and command.
```
Comment (plaindocs) [inline pages/pipelines/build_matrix.md.erb:20]:
```suggestion
You can use a Build Matrix to expand a single step into three steps by interpolating the matrix values into [command step](/docs/pipelines/command-step) attributes such as [environment variables](/docs/pipelines/environment-variables), labels and commands. Interpolation of matrix values inside plugin attributes is not supported.
```

File: pages/pipelines/build_matrix.md.erb [hunk 2]
```diff
@@ -17,19 +17,24 @@ steps:
...
 
 ```yaml
 steps:
   - label: "{{matrix}} build"
     command: "GOOS={{matrix}} go build"
+    env:
```
Comment (plaindocs) [inline pages/pipelines/build_matrix.md.erb:26]: I like this addition!

File: pages/pipelines/build_matrix.md.erb [hunk 3]
```diff
@@ -17,19 +17,24 @@ steps:
...
     matrix:
       - "darwin"
       - "Linux"
       - "Windows"
 ```
 {: codeblock-file="pipeline.yml"}
 
+<section class="Docs__note">
+  <p>Currently interpolation of matrix values inside plugin section is not supported </p>
+</section>
```
Comment (plaindocs) [inline pages/pipelines/build_matrix.md.erb:37]:
```suggestion
```

I think we can just add this to the previous paragraph.

**Review by plaindocs — APPROVED** [review]
Thanks @nithyaasworld this is great.

**Comment by pzeballos** [comment]
It was @nsuma8989 @plaindocs 😉

**Comment by plaindocs** [comment]
Thanks @pzeballos !!

Thanks and sorry @nsuma8989 !!

And thanks @nithyaasworld for good measure !!

**Pull request merged**